cmake/dependencies.cmake

```diff
@@ -8,21 +9,31 @@ pkg_check_modules(netcdff IMPORTED_TARGET REQUIRED netcdf-fortran)
 ################################################################################
 # json-fortran library
 
-find_path(JSON_INCLUDE_DIR json_module.mod
-  DOC "json-fortran include directory (must include json_*.mod files)"
-  PATHS
-    $ENV{JSON_FORTRAN_HOME}/lib
-    /opt/local/lib
-    /usr/local/lib
-    /usr/local/lib64)
-find_library(JSON_LIB jsonfortran
-  DOC "json-fortran library"
-  PATHS
-    $ENV{JSON_FORTRAN_HOME}/lib
-    /opt/local/lib
-    /usr/local/lib
-    /usr/local/lib64)
-include_directories(${JSON_INCLUDE_DIR})
+if(NOT ENABLE_YAML)
+  find_path(JSON_INCLUDE_DIR json_module.mod
+    DOC "json-fortran include directory (must include json_*.mod files)"
+    PATHS
+      $ENV{JSON_FORTRAN_HOME}/lib
+      /opt/local/lib
+      /usr/local/lib
+      /usr/local/lib64)
+  find_library(JSON_LIB jsonfortran
+    DOC "json-fortran library"
+    PATHS
+      $ENV{JSON_FORTRAN_HOME}/lib
+      /opt/local/lib
+      /usr/local/lib
+      /usr/local/lib64)
+  include_directories(${JSON_INCLUDE_DIR})
+endif()
+
+################################################################################
+# yaml-cpp lirary
+
+if(ENABLE_YAML)
+  find_package(yaml-cpp REQUIRED)
+  include_directories(${YAML_CPP_INCLUDE_DIR})
```

Comment:
You could change this to be

```
FetchContent_Declare(yaml
      GIT_REPOSITORY https://github.com/jbeder/yaml-cpp.git
      GIT_TAG 0.8.0
      FIND_PACKAGE_ARGS NAMES yaml-cpp 
  )
  FetchContent_MakeAvailable(yaml)
```

Then it will look for yaml-cpp with find package first and if it can't be found it'll download a version from github. 
Also, it would be better to remove `include_directories(${YAML_CPP_INCLUDE_DIR})` and instead link yaml-cpp to musica like this, which should add the include path to musica

```
 target_link_libraries(musica 
    PRIVATE 
      yaml-cpp::yaml-cpp
  )
```

Reply:
done!